src/index/index.h

```diff
@@ -28,18 +28,33 @@ typedef struct {
 	EntityID edge_id;
 } EdgeIndexKey;
 
+typedef struct {
+	char *name;
+	Attribute_ID id;
+	double weight;
+	bool nostem;
+	char *phonetic;
+} IndexField;
```

Comment:
Please add comments, explaining the purpose of each `field`

Reply:
Also do we need this struct in the header file? if no one excepts `index.c` uses it we should not expose it.